neuralmonkey/nn/projection.py

```diff
@@ -0,0 +1,47 @@
+"""
+This module implements various types of projections.
+"""
+#tests: lint
+import tensorflow as tf
+
+def linear(inputs, size, scope="LinearProjection"):
```

Comment:
I would keep the non-linearity outside the call of this function. Or, create something like `nn.projection.nonlinear`, which would use `linear` and append the activation... But I think the code will be more readable with `tf.tanh(linear(tensor, size))` instead of `project(tensor, size, activation=tf.tanh)`